api/v1alpha1/etcdpeer_types.go

```diff
@@ -7,10 +7,48 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// InitialClusterMemeber describes a single member of the initial cluster.
+type InitialClusterMember struct {
+	// Name is the initial name of the peer, which is set into etcd at the
+	// end of this process
+	Name string `json:"name"`
+
+	// Host forms part of the Advertise URL - the URL at which this peer can
+	// be contacted. The port and scheme are hardcoded.
+	Host string `json:"host"`
```

Comment:
do you think it could be nicer to make this field a bit more generic? i.e. something like

```
    ...

    Host InitialClusterMemberResolver `json:"resolver"`
}

type InitialClusterMemberResolver struct {
    Host *ClusterMemberHostResolver `json:"host"`

    // maybe we'll want something like this in the future
    Selector *ClusterMemberSelectorResolver `json:"selector"`
}

ClusterMemberHostResolver{
    Host string
    Port intstr.IntOrString
    Scheme corev1.URIScheme
}

ClusterMemberSelectorResolver{
    Selector map[string]string
}
```

or am i trying to over engineer this?

Reply:
> or am i trying to over engineer this?

I would say that we probably don't need this level of complexity right now, and probably through beta.

Plus I don't think resolving by a selector would actually work. The selected pod has to have a resolvable DNS name so that `etcd` running in the Pod can actually hit it. This ultimately needs to be a DNS name that we can pass in using the `ETCD_INITIAL_CLUSTER` environment variable.

Reply:
> Plus I don't think resolving by a selector would actually work. 

you could use the selector the find the pod IPs & pass those to `ETCD_INITIAL_CLUSTER`, right?

> I would say that we probably don't need this level of complexity right now, and probably through beta.

i think you're right, however we should make sure we look over the API for things like this before we lock in to `v1`.

Reply:
> you could use the selector the find the pod IPs & pass those to `ETCD_INITIAL_CLUSTER`, right?

You'd end up with a "chicken & egg" problem. You need to select the pod to get the pod's IP, but the IP doesn't get assigned until the pod is created, and you can't create the pod until you have the full list of IP addresses to pass down into the environment variable.

You could do some kind of init container, where you make the pod, block, wait for the operator to get the IP addresses and pass them to the other pods, then move onto actually starting the etcd container. But even that wouldn't work if a pod happens to restart during this process (and therefore gets a new IP).